### PR TITLE
Add options to build without GTest submodule

### PR DESCRIPTION
**Pull request by binji**
This is used by the Wasm waterfall.

### CODE REVIEW

**Comment by binji** [comment]
@dschuff FYI

File: CMakeLists.txt [hunk 1]
```diff
@@ -116,8 +116,13 @@ add_executable(sexpr-wasm ${SEXPR_WASM_SRCS})
 add_dependencies(everything sexpr-wasm)
 
 # hexfloat-test
+option(BUILD_TESTS "Build GTest-based tests" ON)
```
Comment (dschuff) [inline CMakeLists.txt]: I don't see anything conditioned on this option other than the error message?

Comment (dschuff) [inline CMakeLists.txt]: oh wait, my bad, the outer if replaces the previous if. So yeah i think this works.

**Comment by dschuff** [comment]
This doesn't quite work for me when I tried it with a no-submodules checkout. Still looking into it.

File: CMakeLists.txt (hunk 1 above)
Comment (dschuff) [inline CMakeLists.txt]: The add_executable on line 136 also needs to be conditioned on BUILD_TESTS

Comment (binji) [inline CMakeLists.txt]: Really? That's strange because it's inside this if already...

Comment (dschuff) [inline CMakeLists.txt]: Oops, I meant 169

Comment (dschuff) [inline CMakeLists.txt]: https://gist.github.com/dschuff/037a8c4e93668e7c98f4

Comment (dschuff) [inline CMakeLists.txt]: FTR, wrapping 168 in `if (BUILD_TESTS)` / `endif()` makes it work.

Comment (binji) [inline CMakeLists.txt]: done.

File: CMakeLists.txt [hunk 2]
```diff
@@ -128,7 +133,10 @@ if (CMAKE_USE_PTHREADS_INIT)
     third_party/gtest/googletest
     third_party/gtest/googletest/include
   )
-  add_executable(hexfloat_test ${HEXFLOAT_TEST_SRCS})
+  # TODO(binji): why is this additional test needed for newer CMake?
+  if (BUILD_TESTS)
```
Comment (dschuff) [inline CMakeLists.txt]: Sorry I mistyped my original comment. This is unnecessary, it's line 172 in the current commit that needs to be guarded. It's outside the big existing `if (BUILD_TESTS)`

Comment (binji) [inline CMakeLists.txt]: oh :)

**Pull request merged**